### ISSUE TITLE
Test suite fails on darwin in Cranelift::spec::skip_stack_guard_page

### ISSUE THREAD

**Issue by nomeata** [bytecodealliance/wasmtime#1556]
Trying to upgrade `wasmtime` in nixpks via https://github.com/NixOS/nixpkgs/pull/85602
 I get the following build error
```
thread 'Cranelift::spec::skip_stack_guard_page' has overflowed its stack
fatal runtime error: stack overflow
error: test failed, to rerun pass '--test wast_testsuites'
Caused by:
  process didn't exit successfully: `/private/tmp/nix-build-wasmtime-0.15.0.drv-0/source/target/debug/deps/wast_testsuites-eaac573cefb8aa83` (signal: 6, SIGABRT: process abort signal)
```
Full log at https://logs.nix.ci/?key=nixos/nixpkgs.85602&attempt_id=83a9c69d-7d43-491e-959c-9d974a859018

**Comment by alexcrichton** [comment]
Is it possible to debug the builders and get perhaps a full stack trace and/or more info for the fault here?

**Comment by nomeata** [comment]
I’ll try to reproduce it locally and see if I can get something

**Comment by Mic92** [comment]
Maybe you can do something like

```nix
{
  preCheck = ''
    export RUST_BACKTRACE=1
  '';
}
```

and re-run ofborg.

**Comment by nomeata** [comment]
Tried that locally, but it didn’t produce any new output

**Comment by nomeata** [comment]
Just for the record: I failed to debug this, and don’t expect I will be able to, it’s out of my expertise.

**Comment by jyn514** [comment]
> Is it possible to debug the builders and get perhaps a full stack trace and/or more info for the fault here?

In general, Rust does not generate backtraces for stack overflows. You would have to use `gdb` on the process, something like this:

```
$ gdb /home/joshua/.local/lib/cargo/target/debug/rcc
Reading symbols from /home/joshua/.local/lib/cargo/target/debug/rcc...done.
warning: Missing auto-load script at offset 0 in section .debug_gdb_scripts
of file /home/joshua/.local/lib/cargo/target/debug/rcc.
Use `info auto-load python-scripts [REGEXP]' to list them.
(gdb) run crash.c
Starting program: /home/joshua/.local/lib/cargo/target/debug/rcc crash.c
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x0000555555e4754b in core::ptr::const_ptr::<impl *const T>::offset (
    self=0x7fffff7ff2a4 "\017\000", count=0)
    at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ptr/const_ptr.rs:160
160	/rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ptr/const_ptr.rs: No such file or directory.
(gdb) where
#0  0x0000555555e4754b in core::ptr::const_ptr::<impl *const T>::offset (
    self=0x7fffff7ff2a4 "\017\000", count=0)
    at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ptr/const_ptr.rs:160
#1  0x0000555555e47514 in core::ptr::const_ptr::<impl *const T>::add (
    self=0x7fffff7ff2a4 "\017\000", count=0)
    at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ptr/const_ptr.rs:399
... etc ...
```

Since it's in the test suite, you'll use the binary that `cargo test` generates, something like this:
```
$ cargo test --lib noop
    Finished test [unoptimized + debuginfo] target(s) in 0.08s
     Running /home/joshua/.local/lib/cargo/target/debug/deps/rcc-d000a597a330c054

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 117 filtered out

$ gdb /home/joshua/.local/lib/cargo/target/debug/deps/rcc-d000a597a330c054
Reading symbols from /home/joshua/.local/lib/cargo/target/debug/deps/rcc-d000a597a330c054...done.
(gdb)
```

**Comment by nomeata** [comment]
Reportedly fixed in 0.16, according to https://github.com/ggreif/nixpkgs/commit/d89ae377d4b57415b1b00f2aa8e5a1655a1724b9
